### ISSUE TITLE
[ios] app immediataly drop 60 to 40 FPS when move the map with PROVIDER_GOOGLE

### ISSUE THREAD

**Issue by khanh21011999** [react-native-maps/react-native-maps#4663]
### Summary

I'm trying to run the template app from react-native, but everytime, I move the map with google map, it immediately drop from 40 or below, this happen on ios real device, any fix for this, this just a template code, I do not add anything

### Reproducible sample code

```TSX
import MapView, {PROVIDER_GOOGLE} from 'react-native-maps'; // remove PROVIDER_GOOGLE import if not using Google Maps
import {View, StyleSheet} from 'react-native';
import React, {useMemo} from 'react';
const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    height: 400,
    width: 400,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  map: {
    ...StyleSheet.absoluteFillObject,
  },
});

const region = {
  latitude: 37.78825,
  longitude: -122.4324,
  latitudeDelta: 0.015,
  longitudeDelta: 0.0121,
};

export default () => {
  const mapComponent = useMemo(() => {
    return (
      <MapView
        provider={PROVIDER_GOOGLE}
        style={styles.map}
        region={{
          latitude: 37.78825,
          longitude: -122.4324,
          latitudeDelta: 0.015,
          longitudeDelta: 0.0121,
        }}
        loadingEnabled={true}
        showsUserLocation={true}
        showsMyLocationButton={false}
        showsPointsOfInterest={false}
        showsCompass={false}
        showsBuildings={false}
        showsTraffic={false}
        showsIndoors={false}
      />
    );
  }, [region]);

  return <View style={styles.container}>{mapComponent}</View>;
};
```

### Steps to reproduce

npx react-native run-ios device="your real device"

### Expected result

Map should run smooth 60PFS all the time

### Actual result

It drop to below 40FPS immediately and few second of stand still, it comeback 

### React Native Maps Version

1.4.0

### What platforms are you seeing the problem on?

iOS (Google Maps)

### React Native Version

0.71

### What version of Expo are you using?

Not using Expo

### Device(s)

Iphone XR

**Comment by yeesiang95** [comment]
Same!

**Comment by alperenkarip** [comment]
i have the same problem, fps drops when i scroll and release the map. I've been searching for a week but couldn't find a solution. even though i made a lot of performance fixes..

**Comment by alperenkarip** [comment]
https://user-images.githubusercontent.com/5768579/233502748-fdb1eb54-3d77-4142-b929-ac679af6217c.MOV

**Comment by alperenkarip** [comment]
@khanh21011999 did you solve it ?

**Comment by yeesiang95** [comment]
i'm facing the same issue

**Comment by Santosl2** [comment]
Any updates??

**Comment by skaionedev** [comment]
Experiencing the same issue. FPS drops both on simulator and on a real device(13PRO) when using Google Provider

**Comment by EthanDM** [comment]
I am running into this issue too. Google Maps seems a bit laggy/jerky, especially when zoomed in and throwing the map around. Apple Maps is very smooth.

I investigated further and tested apps that use Google Maps (AirBnB, DoorDash, etc.) and also seem to have the lag. The native Google Maps app feels smoother.

Granted both AirBnB and DoorDash _may_ be using this package or some React Native implementation (not sure) but the point stands. Especially surprising in apps such as those. There is a noticeable lag in those apps and perhaps made worse due to the callouts/data they have on them.

Since Google Maps vs Apple Maps isn't a big dealbreaker for me, I may just use Apple Maps to provide a smooth experience but prefer Google Maps.

**Comment by CalNicklin** [comment]
See my solution to this:
https://medium.com/@calpnicklin/60fps-react-native-google-maps-on-ios-563b56248e2f

**Comment by khanh21011999** [comment]
> See my solution to this: https://medium.com/@calpnicklin/60fps-react-native-google-maps-on-ios-563b56248e2f

Thank for solution, but I think it happen also if no marker at all, I think we can improve this one

**Comment by khanh21011999** [comment]
> @khanh21011999 did you solve it ?

Feel like your UI is very smooth

**Comment by yeesiang95** [comment]
i also looking for solution, empty map but fps drop when i scroll

**Comment by CalNicklin** [comment]
> > See my solution to this: https://medium.com/@calpnicklin/60fps-react-native-google-maps-on-ios-563b56248e2f
> 
> Thank for solution, but I think it happen also if no marker at all, I think we can improve this one

Ahh, yes you're right. No markers. Have you tried this without your useMemo?

**Comment by CalNicklin** [comment]
> > > See my solution to this: https://medium.com/@calpnicklin/60fps-react-native-google-maps-on-ios-563b56248e2f
> > 
> > 
> > Thank for solution, but I think it happen also if no marker at all, I think we can improve this one
> 
> Ahh, yes you're right. No markers. Have you tried this without your useMemo? Insert your MapView into the return without memoizing it first.

**Comment by khanh21011999** [comment]
> > > See my solution to this: https://medium.com/@calpnicklin/60fps-react-native-google-maps-on-ios-563b56248e2f
> > 
> > 
> > Thank for solution, but I think it happen also if no marker at all, I think we can improve this one
> 
> Ahh, yes you're right. No markers. Have you tried this without your useMemo?

Not work, above example is pure react-native project without any expensive calculation

**Comment by MariuzM** [comment]
Still same issue feels like app is running in 30fps but stats show 60

**Comment by unkn-wn** [comment]
Still nothing?

**Comment by K4CZP3R** [comment]
I've noticed when I'm running reanimated animation, mapView does not feel like 40fps anymore.

[link to the video](https://www.youtube.com/watch?v=tOxkBPeQU9Y)

Watch it in 60 fps mode to see the difference.

**Comment by MariuzM** [comment]
> I've noticed when I'm running reanimated animation, mapView does not feel like 40fps anymore.
> 
> [link to the video](https://www.youtube.com/watch?v=tOxkBPeQU9Y)
> 
> Watch it in 60 fps mode to see the difference.

Hm so you run Reanimated Animation on top of Map and just that it runs in 60fps? So does that mean Reanimated is triggering something in Native to allow for faster frames but MapView on its own does not do that?

**Comment by K4CZP3R** [comment]
Yup, but I could not find how reanimated triggers RN to render more frames.

**Comment by Iuriy-Budnikov** [comment]
:(

**Comment by petemusicpancake** [comment]
still an issue

**Comment by dungphung** [comment]
Still facing this issue

**Comment by MariuzM** [comment]
bump

**Comment by MariuzM** [comment]
Still relevant

**Comment by celian-riboulet** [comment]
Still relevant

**Comment by datngotan-aavn** [comment]
Still relevant

**Comment by awj5** [comment]
Still relevant

**Comment by niktosh** [comment]
Still relevant

**Comment by NicholasBoccuzzi** [comment]
Bump!

**Comment by margox** [comment]
Still relevant

**Comment by slacki123** [comment]
still relevant

**Comment by salah-ghanim** [comment]
commenting still relevant doesn't really make sense here, the library is using Fabric for react-native-maps GoogleMaps for iOS so we have the best possible performance.

If you still see this as an issue I would recommend to first compare google-maps frame rate on iOS on it's own without react-native and let's see how much over head this lib adds.

for now I'm closing this because the work that was done to migrate to Fabric is significant enough to impact improve this issue within the limitations of react-native.
